src/main/java/org/spongepowered/mod/mixin/core/event/player/MixinEventPlayerInteractBlock.java

```diff
@@ -92,4 +96,15 @@ public Direction getSide() {
         return Optional.fromNullable(new Cause(null, this.entityPlayer, null));
     }
 
+    @Override
+    public void setClickedPosition(Vector3d clickedPosition) {
+        this.clickedPosition = Optional.of(VecHelper.toVector(this.pos).toDouble().add(clickedPosition));
+    }
+
+    @Override
+    public Optional<Vector3d> getClickedPosition() {
+        if (!clickedPosition.isPresent())
+            clickedPosition = Optional.of(VecHelper.toVector(this.pos).toDouble());
```

Comment:
This should remain absent if the position is not available. The block location is already known due to the event extending `BlockEvent`